### PR TITLE
R2 13 1

### PR DESCRIPTION
**Pull request by tomkordic**
Added support for ExoPlayer 2.13.1, API same, lots of copy paste.

### CODE REVIEW

File: automatedtests/src/androidTest/java/com/mux/stats/sdk/muxstats/automatedtests/ui/SimplePlayerBaseActivity.java [hunk 1]
```diff
@@ -219,8 +219,7 @@ public boolean waitForPlaybackToStart(long timeoutInMs) {
   }
 
   public void waitForPlaybackToStartBuffering() {
-    if (player.getPlaybackState() == Player.STATE_READY &&
-        player.getPlayWhenReady()) {
+    if (!muxStats.isPaused()) {
```
Comment (tomkordic) [inline automatedtests/src/androidTest/java/com/mux/stats/sdk/muxstats/automatedtests/ui/SimplePlayerBaseActivity.java:222]: We can not access Player from different Thread it must be the UI thread, this is a fix for the issue.

File: demo/src/r2_10_6/java/com/google/android/exoplayer2/demo/SampleChooserActivity.java [hunk 2]
```diff
@@ -231,8 +233,13 @@ private static boolean isNonNullAndChecked(@Nullable MenuItem menuItem) {
...
         try {
-          readSampleGroups(
-              new JsonReader(new InputStreamReader(inputStream, StandardCharsets.UTF_8)), result);
+          if (VERSION.SDK_INT >= VERSION_CODES.KITKAT) {
+            readSampleGroups(
+                new JsonReader(new InputStreamReader(inputStream, StandardCharsets.UTF_8)), result);
+          } else {
+            readSampleGroups(
+                new JsonReader(new InputStreamReader(inputStream)), result);
+          }
```
Comment (tomkordic) [inline demo/src/r2_10_6/java/com/google/android/exoplayer2/demo/SampleChooserActivity.java:242]: I do not know at which point we enabled the lint checks again,, but this was causing a build to failed, so this is a quick fix, all though it is not a part of the ticket itself.

File: MuxExoPlayer/src/r2_13_1/java/com/mux/stats/sdk/muxstats/MuxStatsExoPlayer.java [hunk 3]
```diff
@@ -0,0 +1,397 @@
...
+        ((SimpleExoPlayer) player).removeAnalyticsListener(this);
+      } else {
+        player.removeListener(this);
+      }
+    }
+    super.release();
+  }
+
+  // ------BEGIN AnalyticsListener callbacks------
+  @Override
```
Comment (tomkordic) [inline MuxExoPlayer/src/r2_13_1/java/com/mux/stats/sdk/muxstats/MuxStatsExoPlayer.java:101]: This entire file is Exact same copy of it's r2_12_1 version with exception that 
@Override
  public void onAudioSessionId(AnalyticsListener.EventTime eventTime, int audioSessionId) {
  }
  method was removed in r2.13.1 and our SDK was not using it anyway.

File: automatedtests/src/androidTest/java/com/mux/stats/sdk/muxstats/automatedtests/ui/SimplePlayerBaseActivity.java (hunk 1 above)
Comment (skidder) [reply automatedtests/src/androidTest/java/com/mux/stats/sdk/muxstats/automatedtests/ui/SimplePlayerBaseActivity.java:222]: @tomkordic is this the main change needed to support `r2.13.1`? If there were additional changes, would you please call them out.

Comment (tomkordic) [reply automatedtests/src/androidTest/java/com/mux/stats/sdk/muxstats/automatedtests/ui/SimplePlayerBaseActivity.java:222]: This is only automated test related, I did comment here: https://github.com/muxinc/mux-stats-sdk-exoplayer/pull/98#discussion_r596475930 this is the only relevant change to the MuxStats, basically the changes in the API are minimal, most of this version is just copy of version 2.12.1

**Review by skidder — APPROVED** [review]

File: MuxExoPlayer/build.gradle [hunk 4]
```diff
@@ -130,6 +142,12 @@ dependencies {
     r2_12_1Api 'org.checkerframework:checker-qual:2.5.2'
     r2_12_1_adsApi 'com.google.android.exoplayer:exoplayer:2.12.1'
     r2_12_1_adsApi 'org.checkerframework:checker-qual:2.5.2'
+
```
Comment (jsanford8) [inline MuxExoPlayer/build.gradle]: I'd remove the extra newlines here

File: demo/build.gradle [hunk 5]
```diff
@@ -138,4 +144,11 @@ dependencies {
     r2_11_1_adsImplementation 'com.google.ads.interactivemedia.v3:interactivemedia:3.11.3'
     r2_12_1_adsImplementation 'com.google.android.gms:play-services-ads-identifier:17.0.0'
     r2_12_1_adsImplementation 'com.google.ads.interactivemedia.v3:interactivemedia:3.20.1'
+    r2_12_1Api "com.google.android.gms:play-services-cronet:17.0.0"
+    r2_12_1_adsApi "com.google.android.gms:play-services-cronet:17.0.0"
+
```
Comment (jsanford8) [inline demo/build.gradle:149]: I'd just remove this extra line

**Review by jsanford8 — APPROVED** [review]
Looks like a pretty standard new version from Google where they break the API by removing a method, so we need a whole new flavor. Just a newline comment or two, but I think this is good to go

**Pull request merged**